### PR TITLE
feat(timezone): use timeZone prop to show values

### PR DESCRIPTION
**Pull request by ValeraS**

### CODE REVIEW

**Review by korvin89 — APPROVED** [review]

File: src/components/Calendar/hooks/useCalendarState.ts [hunk 1]
```diff
@@ -19,14 +23,39 @@ const defaultModes: Record<CalendarLayout, boolean> = {
...
+        props.defaultMode ?? minMode,
+        props.onUpdateMode,
+    );
+
     const currentMode = mode && availableModes.includes(mode) ? mode : minMode;
 
+    const defaultTimeZone = useDefaultTimeZone(
+        props.value || props.defaultValue || props.focusedValue || props.defaultFocusedValue,
+    );
+    const inputTimeZone = defaultTimeZone || props.timeZone || 'default';
```
Comment (EgorKluch) [inline src/components/Calendar/hooks/useCalendarState.ts:46]: Maybe you add second argument `defaultValue` to `useDefaultTimeZone`?

```
const inputTimeZone = useDefaultTimeZone(
    props.value || props.defaultValue || props.focusedValue || props.defaultFocusedValue,
    props.timeZone || 'default'
);
```

File: src/components/Calendar/hooks/useCalendarState.ts [hunk 2]
```diff
@@ -48,15 +77,17 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
 
     const focusedDate =
         focusedDateInner ??
-        constrainValue(createPlaceholderValue({timeZone}).startOf(minMode), minValue, maxValue);
+        constrainValue(createPlaceholderValue({timeZone: inputTimeZone}), minValue, maxValue);
 
     if (isInvalid(focusedDate, minValue, maxValue)) {
         // If the focused date was moved to an invalid value, it can't be focused, so constrain it.
-        setFocusedDate(constrainValue(focusedDate, minValue, maxValue));
+        setFocusedDate(constrainValue(focusedDate, minValue, maxValue).timeZone(inputTimeZone));
```
Comment (EgorKluch) [inline src/components/Calendar/hooks/useCalendarState.ts:84]: `setFocusedDate(constrainValue(focusedDate.timeZone(inputTimeZone), minValue, maxValue));`?

File: src/components/Calendar/hooks/useCalendarState.ts [hunk 3]
```diff
@@ -68,16 +99,20 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
...
         timeZone,
         selectDate(date: DateTime, force = false) {
             if (!disabled) {
                 if (!readOnly && (force || this.mode === minMode)) {
-                    const newValue = constrainValue(date, minValue, maxValue);
-                    if (this.isCellUnavailable(newValue)) {
-                        return;
-                    }
-                    setValue(newValue);
+                    this.setValue(date.startOf(minMode));
```
Comment (EgorKluch) [inline src/components/Calendar/hooks/useCalendarState.ts:115]: Why are you apply timeZone [there](https://github.com/gravity-ui/date-components/pull/36/files#diff-ef006b04b74e2f360e93bdff6025b71a1871482c28a3a9ddcb44c27138e2e7e2R89)

But not here?

File: src/components/Calendar/hooks/useCalendarState.ts [hunk 4]
```diff
@@ -194,6 +229,9 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
         isWeekend(date: DateTime) {
             return this.mode === 'days' && [0, 6].includes(date.day());
         },
+        isCurrent(date: DateTime) {
+            return dateTime({timeZone}).isSame(date, this.mode);
```
Comment (EgorKluch) [inline src/components/Calendar/hooks/useCalendarState.ts:233]: [isSame](https://github.com/gravity-ui/date-utils/blob/986b63f1996a86a57ce261739a27eb05a61ae2c4/src/dateTime/dateTime.ts#L234) does not depend on the time zone

File: src/components/Calendar/hooks/useCalendarState.ts [hunk 5]
```diff
@@ -172,10 +207,10 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
             return this.isInvalid(next);
         },
         isSelected(date: DateTime) {
-            return (
-                Boolean(value) &&
-                date.isSame(value ?? undefined, currentMode) &&
-                !this.isCellDisabled(date)
+            return Boolean(
+                value &&
+                    date.isSame(value.timeZone(timeZone), currentMode) &&
```
Comment (EgorKluch) [inline src/components/Calendar/hooks/useCalendarState.ts:212]: [isSame](https://github.com/gravity-ui/date-utils/blob/986b63f1996a86a57ce261739a27eb05a61ae2c4/src/dateTime/dateTime.ts#L234) does not depend on the time zone

**Pull request merged**